src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -215,10 +215,24 @@ private RestClientMethod BuildMethod(Operation operation, string clientName, IRe
                     }
                     else
                     {
-                        constantOrReference = parameter = BuildParameter(requestParameter);
+                        parameter = BuildParameter(requestParameter);
+
+                        if (requestParameter.GroupedBy is RequestParameter groupedByParameter)
+                        {
+                            var groupModel = (ObjectType)_typeFactory.CreateType(groupedByParameter.Schema, false).Implementation;
+                            var property = groupModel.GetPropertyForGroupedParameter(requestParameter);
+
+                            constantOrReference = new Reference($"{groupedByParameter.CSharpName()}.{property.Declaration.Name}", property.Declaration.Type);
+                        }
+                        else
+                        {
+                            constantOrReference = parameter;
+                        }
                     }
 
-                    if (parameter != null && requestParameter.Flattened != true)
+                    if (parameter != null &&
+                        requestParameter.Flattened != true &&
+                        requestParameter.GroupedBy == null)
```

Comment:
This if-statement is the only thing stopping the extraction of creating `Request` as a separate method (for encapsulation and maintenance purposes). I tried this before, but it is a bit tangled. Not for this PR, but it would be nice to have `BuildMethod` just call separate methods for each of the pieces of information it needs.